src/thor/costmatrix.cc

```diff
@@ -703,7 +703,7 @@ void CostMatrix::SetSources(GraphReader& graphreader,
       // Set the initial not_thru flag to false. There is an issue with not_thru
       // flags on small loops. Set this to false here to override this for now.
       BDEdgeLabel edge_label(kInvalidLabel, edgeid, oppedge, directededge, cost, mode_, ec, d, false,
-                             -1);
+                             true, -1);
```

Comment:
This is the pruning state when setting the source and destination locations. There's a comment here which I don't quite understand the reason for:
```
// Set the initial not_thru flag to false. There is an issue with not_thru
// flags on small loops. Set this to false here to override this for now.
```
I didn't find any tests that highlight the issue alluded to here.

Should we follow the same for closure pruning, i.e start in disabled state (`false`)?

Reply:
this is essentially the same as SetOrigin in the bidirectional algorithm we should do the same thing here.  a copy paste wil suffice and the same in the destination (targets) function :smile: